### PR TITLE
[feat][frontend] QR code scanner that accesses webcam, reads QR codes, and outputs data

### PR DESCRIPTION
**Pull request by WilhelmWillie**
Adds a `Scanner` component and a `scan` page that volunteers can access and scan QR codes

Demo video: https://drive.google.com/file/d/1reNHZcbuPEmMELFiSOox3rbaMHYW7HWj/view?usp=sharing

(#211) 

**TO-DO**
* [ ] Optimize QR code scanning... right now it's sluggish and hella unoptimal (#212) 
* [ ] Hit API route on successful scan and show confirmation messages (#213) 
* [ ] Queue scanned IDs into localStorage in case scans fail and internet goes down (#214) 
* [ ] Restrict scan page access to specific user roles (#215)

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -93,6 +93,10 @@ app.prepare().then(() => {
   server.use("/api/person", personRouter);
   server.use("/api/contribution", contributionRouter);
   server.use("/api/live", liveRouter);
+  server.post("/api/scan", (req, res) => {
```
Comment (WilhelmWillie) [inline server.js]: Temporary route, will clean this up with a future PR

**Comment by donutdaniel** [comment]
Looks good, but is it possible to assign a different role like "volunteer"? Admins would get access to the task and judging manager. We can enable access to scanner for both volunteer & admin

File: server.js (hunk 1 above)
Comment (saresend) [reply server.js]: sounds good

File: pages/scan.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+};
+
+Scan.getInitialProps = async ctx => {
+  const { req } = ctx;
+
+  const profile = await getProfile(req);
+
+  // Null profile means user is not logged in, and this is only relevant for admins
+  if (!profile || profile.role !== "admin") {
+    handleLoginRedirect(req);
```
Comment (saresend) [inline pages/scan.tsx]: Add an issue to make sure we're dispatching appropriately depending on the user role

Comment (WilhelmWillie) [reply pages/scan.tsx]: just wrote up #224

**Comment by WilhelmWillie** [comment]
@donutdaniel def, wrote up #224 to make sure we're mindful of that and take care of that eventually

**Pull request merged**